addons/mrp/wizard/mrp_batch_produce.py

```diff
@@ -49,7 +49,10 @@ def _compute_production_text_help(self):
             for move_raw in wizard.production_id.move_raw_ids:
                 if move_raw.product_id.tracking == "none":
                     continue
-                text += wizard.component_separator + move_raw.product_id.display_name
+                if wizard.component_separator and move_raw.product_id.display_name:
+                    text += wizard.component_separator + move_raw.product_id.display_name
+                else:
+                    text += move_raw.product_id.display_name
```

Comment:
Instead, you should set an empty string when the component_separator is not available in the wizard.

```suggestion
               text += (wizard.component_separator or '') + move_raw.product_id.display_name
```